### PR TITLE
std.string.entab_alias_fix

### PR DESCRIPTION
**Pull request by burner**
make entab and entabber accept custom types with an alias this to a string
member of member function

@MartinNowak I created this against master as the 2.069 beta was already created. Input?

### CODE REVIEW

**Comment by burner** [comment]
@rainers updated this one two

File: std/string.d [hunk 1]
```diff
@@ -4153,7 +4154,7 @@ unittest
         $(LREF entab)
   +/
 auto entabber(Range)(Range r, size_t tabSize = 8)
-    if (isForwardRange!Range)
+    if (isForwardRange!Range && isSomeChar!(ElementEncodingType!Range))
```
Comment (rainers) [inline std/string.d]: Is this additional constraint necessary? It wasn't enforced before (and the doc doesn't mention it), but I also don't find any unittests that actually run the rangified functions on ranges that are not strings...

Comment (burner) [inline std/string.d]: I think I misunderstand.

what do you mean by "ranges that are not strings"? 

on line 4471 ranges are passed to entabber

Comment (rainers) [inline std/string.d]: Sorry, I must have missed the byChar suffix.

Comment (burner) [inline std/string.d]: No problem. I think you had a second part to your question.

> Is this additional constraint necessary?

I think the isSomeChar constraint is necessary as we are trying to change character.

**Comment by burner** [comment]
@rainers anything else missing?

**Comment by rainers** [comment]
LGTM

**Comment by rainers** [comment]
Auto-merge toggled on

**Comment by burner** [comment]
thanks

**Pull request merged**